### PR TITLE
[Chat] Add unit tests for Mention Popover position

### PR DESCRIPTION
**Pull request by Leah-Xia-Microsoft**
# What

Added unit tests for Mention Popover position

# Why

Increase tests coverage for the mention feature

# How Tested

Ran tests on beta and stable flavors

# Process & policy checklist

- [x] I have updated the project documentation to reflect my changes if necessary.
- [x] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-components/src/components/MentionPopover.test.tsx [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+      bottom: 0,
+      left: 0
+    };
+
+    renderMentionPopoverComponent(targetRect, targetPositionOffset, 'above');
+
+    const mentionPopover = await screen.findByTestId(mentionSuggestionListContainerId);
+    const elements = document.getElementsByClassName(mentionPopover.className);
+    const style = window.getComputedStyle(elements[0]);
+    expect(style.position).toBe('absolute');
```
Comment (vhuseinova-msft) [inline packages/react-components/src/components/MentionPopover.test.tsx:81]: I'm wondering if it worth to have snapshop tests for this functionality

Comment (Leah-Xia-Microsoft) [reply packages/react-components/src/components/MentionPopover.test.tsx:81]: > I'm wondering if it worth to have snapshop tests for this functionality

Chatted with Calling team, we need to do a lot of setup to be able to add snapshots for component only features. We'll focus on unit test for Public Preview.

File: packages/react-components/src/components/MentionPopover.tsx [hunk 2]
```diff
@@ -299,6 +300,7 @@ export const _MentionPopover = (props: _MentionPopoverProps): JSX.Element => {
   return (
     <div ref={popoverRef}>
       <Stack
+        data-testid={testIds.mentionSuggestionListContainer}
```
Comment (emlynmac) [inline packages/react-components/src/components/MentionPopover.tsx:303]: We should not have test code populating shipped components

Comment (Leah-Xia-Microsoft) [reply packages/react-components/src/components/MentionPopover.tsx:303]: > We should not have test code populating shipped components

We need the testid as  query when there's no display value of the component. https://testing-library.com/docs/queries/about

Adding test id to the shipped components should be normal. We have `data-ui-id` everywhere which is used for end to end testing. This data-testid is for unit testing.

**Review by vhuseinova-msft — APPROVED** [review]

**Review by emlynmac — APPROVED** [review]

**Pull request merged**